### ISSUE TITLE
Fix S1172: Rule should also ignore the new Main syntax

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#1024]
### Description

The rule doesn't report when the code doesn't use `args` from Main method.

### Repro steps

```
using System;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Program
    {
        static async Task Main(string[] args)
        {
            Console.WriteLine("Test");
        }
    }
}
```

### Expected behavior

The rule doesn't raise an issue when not using `args`.

### Actual behavior

The rule raises an issue.
